cdap-ai-ext-vertexai/src/main/java/io/cdap/cdap/ai/ext/vertexai/VertexAIProvider.java

```diff
@@ -53,18 +61,27 @@ public void initialize(AIProviderContext context) throws Exception {
     this.model = new GenerativeModel(conf.getModelName(), vertexAI);
   }
 
+  private Content createContent(String text) {
+    return Content.newBuilder()
+        .addParts(Part.newBuilder().setText(text).build())
+        .setRole("USER")
+        .build();
+  }
+
   /**
-   * Summarizes the application details in the specified format using Vertex AI. Currently, this
-   * method is not implemented and returning a null string.
+   * Summarizes the application details in the specified format using Vertex AI.
    *
    * @param applicationDetail Details of the application to be summarized.
    * @param format The format in which the summary should be provided.
-   * @return This method currently returning a null string.
+   * @return This method will be returning a string which is summary in required format.
    */
   @Override
   public String summarizeApp(ApplicationDetail applicationDetail, String format) {
     try {
-      GenerateContentResponse response = model.generateContent(conf.getPrompt().getPipelineMarkdownSummary());
+      String pipelineDetail = GSON.toJson(applicationDetail);
+      String prompt = conf.getPrompt().getPipelineMarkdownSummary();
+      String payload = pipelineDetail + "\n" + prompt;
+      GenerateContentResponse response = model.generateContent(createContent(payload));
       return ResponseHandler.getText(response);
     } catch (IOException ex) {
       ex.printStackTrace();
```

Comment:
Please use logger instead of printStackTrace

Reply:
Also, better not to use `null` as the return value. Use `Optional<String>` instead if need to distinguish between no summary vs empty summary. Or otherwise, just use empty string.

Reply:
Did the changes in PR  (https://github.com/cdapio/cdap/pull/15680)